RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromZSRawProducer.cc

```diff
@@ -0,0 +1,300 @@
+    static void incAbrt() {}
+    static void incClus(int) {}
+    static void printStat() {}
+#endif
+  };
+
+  class SiStripClusterizerFromZSRaw final : public edm::stream::EDProducer<> {
+  public:
+    explicit SiStripClusterizerFromZSRaw(const edm::ParameterSet& conf)
+        : productToken_(consumes<FEDRawDataCollection>(conf.getParameter<edm::InputTag>("ProductLabel"))),
+          doAPVEmulatorCheck_(conf.getParameter<bool>("DoAPVEmulatorCheck")) {
+      produces<edmNew::DetSetVector<SiStripCluster>>();
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+
+      desc.add<edm::InputTag>("ProductLabel", edm::InputTag("rawDataCollector"));
+      desc.add<bool>("DoAPVEmulatorCheck", false);
+
+      descriptions.add("SiStripClustersFromZSRawFacility", desc);
+    }
+
+    void beginRun(const edm::Run&, const edm::EventSetup& es) override { initialize(es); }
+
+    void produce(edm::Event& ev, const edm::EventSetup& es) override {
+      initialize(es);
+
+      // get raw data
+      edm::Handle<FEDRawDataCollection> rawData;
+      ev.getByToken(productToken_, rawData);
+
+      auto output = std::make_unique<edmNew::DetSetVector<SiStripCluster>>();
+      assert(!output->onDemand());
+
+      output->reserve(15000, 32 * 10000);
+
+      run(*rawData, *output);
+      output->shrink_to_fit();
+      COUT << output->dataSize() << " clusters from " << output->size() << " modules" << std::endl;
+      ev.put(std::move(output));
+    }
+
+  private:
+    void initialize(const edm::EventSetup& es);
+
+    void run(const FEDRawDataCollection& rawColl, edmNew::DetSetVector<SiStripCluster>& output);
+
+  private:
+    edm::EDGetTokenT<FEDRawDataCollection> productToken_;
+
+    ClustersFromZSRawAlgorithm clusterizer_;
+
+    // March 2012: add flag for disabling APVe check in configuration
+    bool doAPVEmulatorCheck_;
+  };
+}  // namespace
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+DEFINE_FWK_MODULE(SiStripClusterizerFromZSRaw);
+
+namespace {
+
+  void SiStripClusterizerFromZSRaw::initialize(const edm::EventSetup& es) { clusterizer_.initialize(es); }
+
+  void SiStripClusterizerFromZSRaw::run(const FEDRawDataCollection& rawColl,
+                                        edmNew::DetSetVector<SiStripCluster>& output) {
+    ClusterFiller filler(rawColl, clusterizer_, doAPVEmulatorCheck_);
+
+    // loop over good det in cabling
+    for (auto const& idet : clusterizer_.allDets()) {
+      StripClusterizerAlgorithm::output_t::FastFiller record(output, idet.detId);
+
+      filler.fill(idet, record);
+
+      if (record.empty())
+        record.abort();
+
+    }  // end loop over dets
+  }
+
+  void ClusterFiller::fill(StripClusterizerAlgorithm::Det const& det,
+                           StripClusterizerAlgorithm::output_t::FastFiller& record) {
+    incReady();
+
+    auto idet = record.id();
+
+    COUT << "filling " << idet << std::endl;
```

Comment:
idet is not used other than this cout, perhaps just replace it here with ```record.id()```

Reply:
the compiler will take care of it